File: app/outlet/outletservice.py
from sqlmodel import Session
from models.outletmodels import Outlet
from dto.outletschema import OutletSchema
from config.database import engine


class OutletService:
    def getall():
        db = Session(engine)
        return db.query(Outlet).all()

    def createOutlet(request: OutletSchema):
        db = Session(engine)
        db_create = Outlet(nama=request.nama, alamat=request.alamat, telp=request.telp)
        db.add(db_create)
        db.commit()

        return "Bisa"

    def updateOutlet(id: int, request: OutletSchema):
        db = Session(engine)
        db_update = db.query(Outlet).filter(Outlet.id == id).first()

        db_update.nama = request.nama
        db_update.alamat = request.alamat
        db_update.telp = request.telp

        db.commit()

        return "Outlet"

    def deleteOutlet(id: int):
        db = Session(engine)
        db_update = db.query(Outlet).filter(Outlet.id == id).first()

        db.delete(db_update)
        db.commit()

        return "Delete Outlet"

File: frontend/src/service/outlet.service.js
import AxioInstance from "../helpers/config";

export const OutletService = {
  createOutlet,
  getAllOutlet,
  updateOutlet,
  deleteOutlet,
  getOutletId,
};

function createOutlet(data) {
  return AxioInstance().post("/outlet", data);
}

function getAllOutlet() {
  return AxioInstance().get("/outlet");
}

function updateOutlet(id, data) {
  return AxioInstance().post(`/outlet/${id}`, data);
}

function getOutletId(id) {
  return AxioInstance().get(`/outlet/${id}`);
}

function deleteOutlet(id) {
  return AxioInstance().delete(`/outlet/${id}`);
}
